runtime/lua/vim/fcnotify.lua

```diff
@@ -0,0 +1,224 @@
+-- Create a file watcher, each watcher is identified by the name of the file that it
+-- watches.
+
+local uv = vim.loop
+local Watcher = {}
+Watcher.__index = Watcher
+local WatcherList = {}
+local check_handle = nil
+
+-- Callback for the check handle, checks if there are pending notifications
+-- for any watcher, and handles them as per the value of the `fcnotify`
+-- option.
+local function check_notifications()
+  for _, watcher in pairs(WatcherList) do
+    if watcher.pending_notifs == true then
+      vim.api.nvim_command('checktime '..watcher.bufnr)
```

Comment:
two things:
1. we should check that bufnr is still valid (`vim.api.nvim_buf_is_valid`)
2. Can we use `string.format` as well, I don't like concatting strings and number.